app/models/tweet.rb

```diff
@@ -0,0 +1,17 @@
+class Tweet < ApplicationRecord
+  belongs_to :user, foreign_key: 'id_user'
+  has_many :bookmarks, foreign_key: 'id_tweet'
+  has_many :likes, foreign_key: 'id_tweet'
+
+  validates :id_user, presence: true
+  validates :content, length: { maximum:256}
+  validates :content_presence_for_tweet_or_quote
```

Comment:
This could have been done easier just with a conditional validations like this one
```suggestion
  validates :content, presence: true, unless: :retweet
```